### PR TITLE
LCR Assignment 1 (Sunbal Cheema

### PR DESCRIPTION
**Pull request by sunbalkc**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Adding code to the assignment-1 file
## What did you learn from the changes you have made?
knn classification, model fiting, model evaluation etc
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
n/a
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
n/a
## How were these changes tested?
n/a
## A reference to a related issue in your repository (if applicable)
n/a
## Checklist
- [x ] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by efantinatti** [comment]
Hi Sunbal

# PR Feedback Report

**Student**: Sunbal  
**GitHub Username**: @sunbalkc   
**PR Number**: 1  
**Branch**: assignment-1  
**Status**: INCOMPLETE  
**PR URL**: https://github.com/sunbalkc/LCR/pull/1

## What You Did Well (Hits)

The pull request exists and is accessible. You correctly identified the number of observations, the response variable type, and its levels. The data was standardized and split into training and testing sets correctly. The cross-validation setup was correct, and the model evaluation on test data was performed. The accuracy calculation is correct based on your model.

## Areas for Improvement (Errors)

Missing some outputs and completion.

## Advice and Recommendations

For data inspection, double-check the number of variables in the dataset. Ensure all features are accounted for. When performing GridSearchCV, consider the impact of the random state on reproducibility, especially for train-test split. While not explicitly required for this assignment, setting a random state for KNeighborsClassifier initialization can ensure consistent results across runs. Review the parameter grid and cross-validation setup to ensure the best hyperparameter aligns with the expected value. Experiment with different random states for the train_test_split to see if it affects the best n_neighbors.

## Overall Assessment

The submission demonstrates a good understanding of the assignment requirements, with most steps correctly. Check the missing outputs.

**Comment by sunbalkc** [comment]
Hello I just did Run ALL to make sure all code blocks ran. please review and let me know if any changes required.

Best Regards,
Sunbal

**Comment by efantinatti** [comment]
Hi @sunbalkc. All good. Completed!
